doc/extensions/tutorial.rst

```diff
@@ -136,8 +136,11 @@ in this example, that inherits from CKAN's
 
 .. note::
 
-   Every CKAN plugin class should inherit from
-   :py:class:`~ckan.plugins.core.SingletonPlugin`.
+  Every CKAN plugin class should inherit from
+  :py:class:`~ckan.plugins.core.SingletonPlugin`. 
+  You are not required to follow this step, 
```

Comment:
Can you move this text to its own note under "Each CKAN extension contains one or more plugins that provide the extension’s features." and replace it with:

> You can skip this section if you run the ``ckan generate extension`` command